doc/sphinx/language/core/records.rst

```diff
@@ -1,248 +1,343 @@
+      {* ( @name @field_spec ) }` and projections are defined by case analysis.
+      In the second implementation, records have
+      primitive projections: see :ref:`primitive_projections`.
 
-An alternative syntax for projections based on a dot notation is
-available:
+   During the definition of the one-constructor inductive definition, all
+   the errors of inductive definitions, as described in Section
+   :ref:`gallina-inductive-definitions`, may also occur.
 
-.. coqtop:: all
+Constructing records
+~~~~~~~~~~~~~~~~~~~~
 
-   Eval compute in half.(top).
+   .. insertprodn term_record field_val
 
-.. flag:: Printing Projections
+   .. prodn::
+      term_record ::= %{%| {*; @field_val } {? ; } %|%}
+      field_val ::= @qualid {* @binder } := @term
 
-   This :term:`flag` activates the dot notation for printing.
+   Instances of record types can be constructed using either *record form*
+   (:n:`@term_record`, shown here) or *application form* (see :n:`@term_application`)
+   using the constructor.  The associated record definition is selected using the
+   provided field names or constructor name, both of which are global.
 
-   .. example::
+   In the record form, the fields can be given in any order.  Fields that can be
+   inferred by unification or by using obligations (see :ref:`programs`) may be omitted.
+
+   In application form, all fields of the record must be passed, in order,
+   as arguments to the constructor.
+
+   .. example:: Constructing 1/2 as a record
+
+      Constructing the rational :math:`1/2` using either the record or application syntax:
+
+      .. coqtop:: in
+
+         Theorem one_two_irred : forall x y z:nat, x * y = 1 /\ x * z = 2 -> x = 1.
+         Admitted.
+
+         (* Record form: top and bottom can be inferred from other fields *)
+         Definition half :=
+           {| negative := false;
+              Rat_bottom_nonzero := O_S 1;
+              Rat_irreducible := one_two_irred |}.
+
+         (* Application form: use the constructor and provide values for all the fields
+            in order.  "mkRat" is defined by the Record command *)
+         Definition half' := mkRat true 1 2 (O_S 1) one_two_irred.
+
+Accessing fields (projections)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+   .. insertprodn term_projection term_projection
+
+   .. prodn::
+      term_projection ::= @term0 .( @qualid {? @univ_annot } {* @arg } )
+      | @term0 .( @ @qualid {? @univ_annot } {* @term1 } )
+
+   The value of a field can be accessed using *projection form* (:n:`@term_projection`,
+   shown here) or with *application form* (see :n:`@term_application`) using the
+   projection function associated with the field.  Don't forget the parentheses for the
+   projection form.
+   Glossing over some syntactic details, the two forms are:
+
+   - :n:`@qualid__record.( {? @ } @qualid__field {* @arg })` (projection) and
+
+   - :n:`{? @ } @qualid__field {* @arg } @qualid__record` (application)
```

Comment:
That's strange. In the sphinx rendering, the space before `(application)` is barely perceptible.

Reply:
Non-breaking space to the rescue:

![image](https://user-images.githubusercontent.com/1253341/154818718-cc9c48ab-6f0c-42a4-a9ee-12dea5bf56ee.png)

Reply:
Much better indeed.